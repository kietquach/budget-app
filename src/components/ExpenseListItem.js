import React from "react";
import { removeExpense } from "../actions/expenses";
import { connect } from "react-redux";

const ExpenseListItem = ({ dispatch, id, description, amount, createdAt }) => (
    <div>
        <h3>{description}</h3>
        {amount}, created at: {createdAt}
        <button onClick={() => {
            dispatch(removeExpense({ id }));
        }}>Remove</button>
    </div>
);

export default connect()(ExpenseListItem);