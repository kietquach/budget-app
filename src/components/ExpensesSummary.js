import React from "react";
import { connect } from "react-redux";
import selectExpenses from "../selectors/expenses";
import getExpensesTotal from "../selectors/expenses-total";
import numeral from "numeral";

export const ExpensesSummary = ({ expensesCount, expensesTotal }) => {

    return (
        <div>
            <h1>Viewing {expensesCount} {expensesCount === 1 ? "expense" : "expenses"} totalling {numeral(expensesTotal / 100).format("$0,0.00")}</h1>
        </div>
    );
};

const mapStatetoProps = (state) => {
    const expenses = selectExpenses(state.expenses, state.filters);

    return {
        expensesCount: expenses.length,
        expensesTotal: getExpensesTotal(expenses)
    };
};

export default connect(mapStatetoProps)(ExpensesSummary);