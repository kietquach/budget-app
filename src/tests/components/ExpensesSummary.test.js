import React from "react";
import { shallow } from "enzyme";
import { ExpensesSummary } from "../../components/ExpensesSummary";

test("Should render ExpensesSummary correctly", () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={3} expensesTotal={115195}/>);
    expect(wrapper).toMatchSnapshot();
});

test("Should render ExpensesSummary with 1 expense corrextly", () => {
    const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={195}/>);
    expect(wrapper).toMatchSnapshot();
});