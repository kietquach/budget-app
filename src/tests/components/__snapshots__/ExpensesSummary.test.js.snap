// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ExpensesSummary correctly 1`] = `
<div>
  <h1>
    Viewing 
    3
     
    expenses
     totalling 
    $1,151.95
  </h1>
</div>
`;

exports[`Should render ExpensesSummary with 1 expense corrextly 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1.95
  </h1>
</div>
`;
